### PR TITLE
Spec: Allow time to be `24:00`

### PR DESCRIPTION
**Pull request by jotaen**
Specification adjustment to account for https://github.com/jotaen/klog/issues/130; based on https://github.com/jotaen/klog/pull/133.

Expand spec to allow the `24:00` time. Effectively, the following cases are allowed additionally:

- `<24:00`, `<12:00pm` (interpreted as `0:00`)
- `24:00`, `12:00pm` (interpreted as `0:00>`)

### CODE REVIEW

**Comment by vladdeSV** [comment]
I usually get confused whenever 24:00 appears in any scenario. I think it is a bit odd that there is so much effort put into allowing exactly one (1) more minute to appear as time.

The only time I would feel the need to write down `24:00` is the same as the example you gave, where you work until exactly midnight. Should you work one minute more and you would still have to specify it as `00:01>`.

Furthermore, I feel there are somewhat bigger implications when time shifting comes into play; I do not understand why `<24:00` would valid, but not `<26:00`. Both my examples specify a time after the current time range, then moves it back to the current day. 

The specification explicitly disallows `24:00>`, and I feel the *pureness* of the concepts "time" and "time shifting" gets disturbed by this change. The previously simple rules gets this extra passage in the specification to allow for a very specific scenario.

If I am explaining the rules of klog to someone, I would have to add this very specific part:

> If the hour is 24, only 00 is allowed as the minute. You are not allowed to time shift forward 24:00, but you can time shift it backwards, which is the same as writing 00:00"

**tl;dr**: i don't really like this change

**Comment by jotaen** [comment]
Thanks for your critical feedback.

On the one hand, I share your reservations about `24:00` being a somewhat illogical exception. On the other hand, it’s a [commonly used and understood practice](https://en.wikipedia.org/wiki/24-hour_clock#Midnight_00:00_and_24:00) (even though it’s not without critique).

To me, it seems there isn’t a clear consensus overall: formal standards, like ISO8601 for example, allowed `24:00:00` until its 2019 revision, then they forbade it to avoid potential ambiguity. Programming libraries, such as Javascript’s `Date()` or [C#’s Nodatime](https://github.com/nodatime/nodatime/issues/198), allow *parsing* 24:00, but they internally “shift” it in the same way as proposed for klog. Other libraries don’t support the `24:00` notation at all.

```js
new Date("2000-01-01T24:00:00.000Z").toISOString()
// ->     2000-01-02T00:00:00.000Z
```

(See [NOTE 1 in the ECMAScript standard](https://262.ecma-international.org/11.0/#sec-date-time-string-format).)

klog is primarily meant to be human-readable/writable, so I’m generally leaning towards allowing exceptions and variations for practical reasons, than towards making the definitions strict and rigorous. The challenge is of course to describe it in a way that it doesn’t make it all complicated and confusing.

I’m not 100% happy with the current phrasing suggestion ([4b98628](https://github.com/jotaen/klog/pull/134/commits/5c8757acae5282a8decacd265203939afdd6ebc1)), so I’m still leaving this open for now and try to improve on it. I’m pretty certain, though, that I want to go forward with this change.
